uefi/src/proto/device_path/mod.rs

```diff
@@ -397,6 +397,35 @@ impl DevicePath {
         total_size_in_bytes
     }
 
+    /// Calculate the size in bytes of the entire `DevicePath` starting
+    /// at `bytes`. This adds up each node's length, including the
+    /// end-entire node.
+    ///
+    /// # Errors
+    ///
+    /// The [`ByteConversionError::InvalidLength`] error will be returned
+    /// when the length of the given bytes slice cannot contain the full
+    /// [`DevicePath`] represented by the slice.
+    fn size_in_bytes_from_slice(bytes: &[u8]) -> Result<usize, ByteConversionError> {
+        let mut ptr = bytes;
```

Comment:
nit: this isn't really a pointer in the traditional sense. You could instead make `bytes` mutable (`mut bytes: &[u8]`), and then change `ptr = &ptr[node_size_in_bytes..];` to `bytes = &bytes[node_size_in_bytes..];`

Reply:
wouldn't `bytes = &bytes[node_size_in_bytes..];` modify the caller's `bytes` as well? I can save the initial value of bytes then reassign it to that before returning, or perhaps rename `ptr` to something more apt, say `start_of_node`

Reply:
It won't modify the callers bytes, its just changing what the local variable points at. Note that it's not the same as `bytes &mut [u8]`, which would allow you to modify the data, and it's not the same as `bytes: &mut &[u8]`, which would allow you to modify the caller's reference to the data.